### PR TITLE
Python 3 - pathod.language.generators

### PR DESCRIPTION
**Pull request by dufferzafar**
Received a huge amount of help from the awesome folks at #python IRC :)

### CODE REVIEW

File: pathod/language/generators.py [hunk 1]
```diff
@@ -37,14 +37,19 @@ def __getitem__(self, x):
...
         return "'transform(%s)'" % self.gen
 
 
+def rand_byte(chars):
+    """
+        Return a random character as byte from a charset.
+    """
+    # bytearray has consistent behaviour on both Python 2 and 3
+    # while bytes does not
+    return bytes(bytearray([random.choice(chars)]))
```
Comment (mhils) [inline pathod/language/generators.py]: I really appreciate the work that went into this, nonetheless let me propose to simplify it by keeping in mind that we want to ditch Python 2 completely once your awesome job is done here:
```
if six.PY2:
    return random.choice(chars)
return bytes([chars])
```

File: pathod/language/generators.py [hunk 2]
```diff
@@ -70,17 +73,17 @@ class FileGenerator(object):
...
 
     def __getitem__(self, x):
-        return self.map.__getitem__(x)
-
-    def __getslice__(self, a, b):
-        return self.map.__getslice__(a, b)
+        if isinstance(x, slice):
+            return self.map.__getitem__(x)
+        # A slice of length 1 returns a byte object (not an integer)
+        return self.map.__getitem__(slice(x, x+1 or self.map.size()))
```
Comment (mhils) [inline pathod/language/generators.py]: I don't understand the `or self.map.size()` part here - what is this about?

Comment (dufferzafar) [inline pathod/language/generators.py]: When `x = -1`, and we want the last character.

Comment (mhils) [inline pathod/language/generators.py]: Ok. But this won't work for `x = -2`, will it? Shall we just use `bytes([chars])` (+ six.PY2) here as well?

Comment (dufferzafar) [inline pathod/language/generators.py]: Um, it does work for -2. Just ran the checks with a bunch of negative numbers.

Comment (mhils) [inline pathod/language/generators.py]: Oh wow, you're absolutely right, we only need to work around x=0. Scratch my last comment!

**Comment by mhils** [comment]
Let's fix https://github.com/mitmproxy/mitmproxy/pull/1195#discussion_r65497564 + the linter and then merge this ASAP! :smiley:

**Comment by mhils** [comment]
I fixed the lint error and merged this. Really great work! :smiley: :cake:

**Pull request merged**